DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -0,0 +1,465 @@
+  h_mass_HLT_fail_forward = ibooker_.book2D("h_mass_HLT_fail_forward", "h_mass_HLT_fail_forward", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+
+  h_mass_SIT_pass_central = ibooker_.book2D("h_mass_SIT_pass_central", "h_mass_SIT_pass_central", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+  h_mass_SIT_pass_forward = ibooker_.book2D("h_mass_SIT_pass_forward", "h_mass_SIT_pass_forward", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+  h_mass_SIT_fail_central = ibooker_.book2D("h_mass_SIT_fail_central", "h_mass_SIT_fail_central", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+  h_mass_SIT_fail_forward = ibooker_.book2D("h_mass_SIT_fail_forward", "h_mass_SIT_fail_forward", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+
+  h_mass_Sta_pass_central = ibooker_.book2D("h_mass_Sta_pass_central", "h_mass_Sta_pass_central", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+  h_mass_Sta_pass_forward = ibooker_.book2D("h_mass_Sta_pass_forward", "h_mass_Sta_pass_forward", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+  h_mass_Sta_fail_central = ibooker_.book2D("h_mass_Sta_fail_central", "h_mass_Sta_fail_central", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+  h_mass_Sta_fail_forward = ibooker_.book2D("h_mass_Sta_fail_forward", "h_mass_Sta_fail_forward", LumiBin_, LumiMin_, LumiMax_, MassBin_, MassMin_, MassMax_);
+
+  h_npv                   = ibooker_.book2D("h_npv",     "h_npv",     LumiBin_, LumiMin_, LumiMax_, PVBin_, PVMin_, PVMax_);
+  h_yield_Z               = ibooker_.book1D("h_yield_Z", "h_yield_Z", LumiBin_, LumiMin_, LumiMax_);
+
+}
+//
+// -------------------------------------- beginLuminosityBlock --------------------------------------------
+//
+void ZCounting::beginLuminosityBlock(edm::LuminosityBlock const& lumiSeg, edm::EventSetup const& context) 
+{
+  edm::LogInfo("ZCounting") <<  "ZCounting::beginLuminosityBlock" << std::endl;
+}
+
+
+//
+// -------------------------------------- Analyze --------------------------------------------
+//
+//--------------------------------------------------------------------------------------------------
+void ZCounting::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{// Fill event tree on the fly 
+  edm::LogInfo("ZCounting") <<  "ZCounting::analyze" << std::endl;
+
+  //-------------------------------
+  //--- Vertex 
+  //-------------------------------
+  edm::Handle<reco::VertexCollection> hVertexProduct;
+  iEvent.getByToken(fPVName_token,hVertexProduct);
+  if(!hVertexProduct.isValid()) return;
+
+  const reco::VertexCollection *pvCol = hVertexProduct.product();
+  const reco::Vertex* pv = &(*pvCol->begin());
+  int nvtx = 0;
+    
+  for(reco::VertexCollection::const_iterator itVtx = pvCol->begin(); itVtx!=pvCol->end(); ++itVtx) {
```

Comment:
you can write this better as
for ( auto const & vex : pvCol) {..
 if ( col.isFake()) ...